include/eve/module/core/regular/impl/simd/riscv/simd_cast.hpp

```diff
@@ -0,0 +1,183 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+namespace eve::detail
+{
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE wide<T, typename N::combined_type>
+                riscv_lmul_ext(wide<T, N> a) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  constexpr auto c = categorize<wide<T, typename N::combined_type>>();
+  static_assert(N::combined_type::value != N::value);
+  constexpr auto in_lmul  = rvv_lmul_v<T, N>;
+  constexpr auto out_lmul = rvv_lmul_v<T, typename N::combined_type>;
+  if constexpr( out_lmul == in_lmul ) return a.storage();
+  else if constexpr( match(c, category::float32) )
+  {
+    if constexpr( out_lmul == 1 ) return __riscv_vlmul_ext_f32m1(a);
+    else static_assert(false, "[eve riscv] Unsupported");
+  }
+  else if constexpr( match(c, category::int32) )
+  {
+    if constexpr( out_lmul == 1 ) return __riscv_vlmul_ext_i32m1(a);
+    else static_assert(false, "[eve riscv] Unsupported");
```

Comment:
Let's replace the word `Unsupported` with `sanity check` - so that the user sees that we have an internal issue, if this happens.

Reply:
Yep, done

Reply:
Even removed for now, as `unit.arch` do not test this functionality. Will add with `sanity check` in future reviews